packages/eslint-plugin-svelte/src/rules/prefer-const.ts

```diff
@@ -0,0 +1,99 @@
+import { createRule } from '../utils';
+import type { TSESTree } from '@typescript-eslint/types';
+
+import { defineWrapperListener, getCoreRule } from '../utils/eslint-core';
+
+const coreRule = getCoreRule('prefer-const');
+
+type Declaration = TSESTree.Expression;
+
+/**
+ * Finds and returns the callee of a declaration node within variable declarations or object patterns.
+ */
+function findDeclarationCallee(node: Declaration) {
+	const { parent } = node;
+	if (parent.type === 'VariableDeclarator' && parent.init?.type === 'CallExpression') {
+		return parent.init.callee;
+	}
+
+	if (
+		parent.type === 'Property' &&
+		parent.parent.type === 'ObjectPattern' &&
+		parent.parent.parent.type === 'VariableDeclarator' &&
+		parent.parent.parent.init?.type === 'CallExpression'
+	) {
+		return parent.parent.parent.init.callee;
+	}
+
+	return null;
+}
+
+/**
+ * Determines if a declaration should be skipped in the const preference analysis.
+ * Specifically checks for Svelte's state management utilities ($state, $props, $derived).
+ */
+function shouldSkipDeclaration(declaration: Declaration | null) {
+	if (!declaration || !declaration.parent) {
+		return false;
+	}
+
+	const callee = findDeclarationCallee(declaration);
+	if (!callee) {
+		return false;
+	}
+
+	if (callee.type === 'Identifier' && ['$state', '$props', '$derived'].includes(callee.name)) {
```

Comment:
Why did you exclude `$state` from rule checking?
I think it's ok to have a `const` variable as long as the user doesn't reassign it in their code 🤔

Reply:
This was part of the discussion in #818. I also proposed having an option that allows the user to decide [which reactive statements](https://github.com/sveltejs/eslint-plugin-svelte/issues/818#issuecomment-2505770979) should be excluded from this rule.

Reply:
I read the thread. So it's ok for you to use `const` for `$state` without reassignment, right?
I'm also ok with `const`, so I think it's better to remove the `$state` branch from the PR and check `$state`.

Reply:
Yes!

Also, what about the option to customize which reactive assignments to ignore?

Reply:
> Also, what about the option to customize which reactive assignments to ignore?

I don't think there's a need for an option for now. If someone needs one, I think they'll suggest it in an issue. If necessary, we'll add the option then.